src/words.ts

```diff
@@ -0,0 +1,161 @@
+type StepVector = Record<string, number>;
+
+// Helper function for the mathematical mod `m` operator to make it return nonnegative integers less than `m`.
+function mod(a: number, m: number): number {
+  return ((a % m) + m) % m;
+}
+
+function sum(array: number[]) {
+  let sum = 0;
+  array.forEach(x => {
+    sum += x;
+  });
+  return sum;
+}
+
+// Helper function that sorts a `Record` in place by key.
+function sortRecordInPlace(record: Record<string, number>) {
+  record = Object.keys(record)
+    .sort((a, b) => (a < b ? -1 : a > b))
+    .reduce((obj, key) => {
+      obj[key] = record[key];
+      return obj;
+    }, {});
+}
+
+export function zeroVector(): StepVector {
+  return {};
+}
+
+export function add(v1: StepVector, v2: StepVector): StepVector {
+  // Clone because we don't want to mutate the input.
+  const result = structuredClone(v1);
```

Comment:
These records are shallow so `result = {...v1}` might be terser and faster.

Reply:
When I call this:
```
// Return the set of letters.
export function letters(sv: StepVector): string[] {
  return Object.keys(sv);
}
```

will the array of letters come out sorted?